JDBC Step2 띠용

jye326 opened the pull request:
안녕하세요 한스! 지난 미션에서 구현한 것과 겹치는 부분이 많아 변경된 부분이 많지 않습니다! 이번에도 자유롭게 리뷰 부탁드릴게요!

jye326 committed:
refactor: CannotExecuteQueryException 추가 및 JdbcTemplate 예외 처리 리팩토링

- CannotExecuteQueryException 클래스 추가
- JdbcTemplate의 executeUpdate 메서드에서 RuntimeException 대신 CannotExecuteQueryException 사용
- DbCleaner 클래스에서 cleanH2 메서드로 네이밍 수정 및 for 문 변수명 가독성 개선
- DbCleaner에서 CannotExecuteQueryException 이용하여 예외 메시지 출력
- UserDaoTest에서 불필요한 for 루프 제거 및 테스트 코드 정리

jye326 committed:
refactor: JdbcExecutionException 추가 및 JdbcTemplate 리팩토링

- JdbcExecutionException으로 클래스명 변경
- JdbcTemplate의 메서드 리팩토링: execute 활용 및 코드 중복 제거
- query 및 update 메서드 추가로 메서드 활용성 향상
- DbCleaner, UserDao 리팩토링: JdbcTemplate의 update 및 query 메서드 사용
- PreparedStatementExecutor 인터페이스 추가 및 JdbcTemplate 개선

jye326 committed:
refactor: DbCleaner의 예외 처리 방식 개선

- System.out.println 대체하여 JdbcExecutionException 재발생 처리
- 예외 발생 시 명시적으로 호출자에게 전달되도록 코드 수정

20HyeonsuLee commented on app/src/test/java/com/techcourse/DbCleaner.java:
```diff
@@ -23,19 +24,19 @@ public void CleanH2() {
+                jdbcTemplate.update("TRUNCATE TABLE " + table);
+            } catch (JdbcExecutionException e) {
+                throw e;
             }
```
요거 한번 확인해보시면 좋을 것 같습니다!

20HyeonsuLee commented on jdbc/src/main/java/com/interface21/jdbc/JdbcExecutionException.java:
```diff
@@ -0,0 +1,7 @@
+package com.interface21.jdbc;
+
+public class JdbcExecutionException extends RuntimeException {
+    public JdbcExecutionException(String message) {
+        super(message);
+    }
```
요것도 확인해보시면 좋을 것 같습니다!

20HyeonsuLee reviewed (approved):
안녕하세요 띠용~

step1때 워낙 잘 구현 해주셨어서 step2에서 딱히 수정할 부분이 없으셨을 것 같아요!

코드레빗 리뷰중 확인해보면 좋은 내용들이 있네요. 
Approve할테니 요것들만 확인해보시고 직접 머지 해주시면 될 것 같습니다!

jye326 committed:
refactor: DbCleaner에서 예외 처리 제거 및 JdbcExecutionException 개선

- DbCleaner에서 불필요한 JdbcExecutionException 활용 예외 처리 코드 제거
- JdbcExecutionException 생성자에 cause 파라미터 추가하여 예외 추적성 강화
- DbCleaner의 SQL 및 코드 가독성 개선

jye326 reviewed (commented):
반영완료했습니다!
